mono/utils/mono-error-internals.h

```diff
@@ -43,10 +43,34 @@ struct _MonoErrorBoxed {
 	MonoImage *image;
 };
 
-#define error_init(error) do {	\
-	((MonoErrorInternal*)(error))->error_code = MONO_ERROR_NONE;	\
-	((MonoErrorInternal*)(error))->flags = 0;	\
-} while (0);
+// FIXME This is duplicated.
+#define ERROR_DECL \
+	MonoError error_value; \
+	MonoError * const error = &error_value; \
+	error_init_internal (error);
+
+#define ERROR_DECL_NAME(x) \
+	MonoError x; \
+	error_init_internal (&x);
+
+// MonoError.hidden_0 is MonoErrorInternal.flags.
+// error_init is only now for compatibility at API boundaries.
+// Internal users should use ERROR_DECL.
+
+#undef error_init
+#define error_init(error) ((error)->error_code = (error)->hidden_0 = 0)
+#undef error_init
+#define error_init(error) // FIXME remove them all
+
+// New macro for new special code. Use with caution.
```

Comment:
I don't understand this comment.

Reply:
The intent is all instances of (whole word) error_init should be removed. However, in order to ease immediate maintenance, adding/removing just a few, I neutered them this way instead. This is a temporary measure, but not a terrible one.

Reply:
Okay we chatted offHub and I understand a bit better.  To summarize: with the changes in this PR, the new semantics for `MonoError` is that you must initialize it at declarations, not at the entry of any function that takes a `MonoError *` argument.
Therefore, there should be vastly fewer calls to `error_init`.
Pretty much the only situation where it will happen (aside from legacy public APIs following the old convention) is in situations where you declare a local `MonoError` to capture some error state that you intend to recover rather than just threading it out to your own caller.

Therefore `error_init` should disappear and `error_init_internal` will take its place in the small number of places where it's needed.

So I guess at this point my one ask is that the comment above `error_init_internal` actually summarize how to use it correctly, rather than just saying "special. use caution" which is not a helpful comment without all of the above context.